test/conversations/Conversations.test.ts

```diff
@@ -148,4 +152,121 @@ describe('conversations', () => {
     expect(aliceConversationsList).toHaveLength(1)
     expect(bobConversationList).toHaveLength(1)
   })
+
+  describe('newConversation', () => {
+    it('uses an existing v1 conversation when one exists', async () => {
+      const aliceConvo = await alice.conversations.newConversation(bob.address)
+      expect(aliceConvo instanceof ConversationV1).toBeTruthy()
+      await aliceConvo.send('gm')
+      const bobConvo = await bob.conversations.newConversation(alice.address)
+      expect(bobConvo instanceof ConversationV1).toBeTruthy()
+    })
+
+    it('continues to use v1 conversation even after upgrading bundle', async () => {
+      const aliceConvo = await alice.conversations.newConversation(bob.address)
+      await aliceConvo.send('gm')
+      expect(aliceConvo instanceof ConversationV1).toBeTruthy()
+      await bob.publishUserContact(false)
+      await sleep(50)
+
+      const aliceConvo2 = await alice.conversations.newConversation(bob.address)
+      expect(aliceConvo2 instanceof ConversationV1).toBeTruthy()
+    })
+
+    it('creates a new V2 conversation when no existing convo and V2 bundle', async () => {
+      await bob.publishUserContact(false)
+      const aliceConvo = await alice.conversations.newConversation(bob.address)
+      expect(aliceConvo instanceof ConversationV2)
+    })
+
+    it('creates a v2 conversation when conversation ID is present', async () => {
+      const conversationId = 'xmtp.org/foo'
+      const aliceConvo = await alice.conversations.newConversation(
+        bob.address,
+        { conversationId, metadata: { foo: 'bar' } }
+      )
+      await sleep(50)
+
+      expect(aliceConvo instanceof ConversationV2)
```

Comment:
Does this actually check the truthiness, or do you need to do invoke `toBeTruthy()` on it?

Reply:
Oops. Fixed